src/order/category/FinDistribLattice.lean

```diff
@@ -0,0 +1,57 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import data.fintype.order
+import order.category.BoundedDistribLattice
+import order.category.FinPartialOrder
+
+/-!
+# The category of finite distributive lattices
+
+This defines `FinDistribLattice`, the category of finite distributive lattices.
+-/
+
+universes u v
+
+open category_theory
+
+/-- The category of finite distributive lattices with bounded lattice morphisms. -/
+structure FinDistribLattice extends BoundedDistribLattice :=
+[is_fintype : fintype X]
+
+namespace FinDistribLattice
+
+instance : inhabited FinDistribLattice := ⟨⟨⟨bool⟩⟩⟩
+instance : has_coe_to_sort FinDistribLattice Type* := ⟨λ X, X.X⟩
+instance (X : FinDistribLattice) : distrib_lattice X := X.is_distrib_lattice
+instance (X : FinDistribLattice) : bounded_order X := X.is_bounded_order
+
+attribute [instance]  FinDistribLattice.is_fintype
+
+/-- Construct a bundled `FinDistribLattice` from a `bounded_order` `distrib_lattice`. -/
+def of (α : Type*) [distrib_lattice α] [bounded_order α] [fintype α] : FinDistribLattice := ⟨⟨α⟩⟩
```

Comment:
Do you want a constructor for `FinDistribLattice` which assumes a `distrib_lattice` which is `fintype`, without the boundedness assumption?

Reply:
It requires `inhabited` or `nonempty` on top, because the empty lattice is not in `FinDistribLattice`. Which one do you want?

Reply:
Nonempty.

Reply:
Done. Should I go as far as deleting the original `of`?